WL-4200 Web Content links in template sites are not duplicated in new…

ouit0408 opened the pull request:
… site.

Checking if the web content url has already been added in new site.if
clause was checking for webcontent urls which never were equal because of
the unique siteId for each site.introduced a constant to check for
equality .

buckett commented on web-impl/impl/src/java/org/sakaiproject/web/impl/WebServiceImpl.java:
```diff
@@ -422,10 +422,12 @@ public void transferCopyEntities(String fromContext, String toContext, List ids)
 								    SitePage p = config.getContainingPage();
+									// Checking if the 'source' property of fromSite and toSite are same by replacing respective siteIds with a constant.
+									String fromContentUrl = contentUrl.replace(fromContext, WEB_CONTENT_URL_PROP);
+									String toContentUrl = config.getPlacementConfig().getProperty(WEB_CONTENT_URL_PROP).replace(toContext,WEB_CONTENT_URL_PROP);
```
Is there the possibility of a NPE here where there isn't a URL set on the tool?

ouit0408 replied:
isn't this method called only for webcontent link type of page  which always will have a url?

buckett replied:
I mean the site it's copying into.

buckett commented on web-impl/impl/src/java/org/sakaiproject/web/impl/WebServiceImpl.java:
```diff
@@ -422,10 +422,13 @@ public void transferCopyEntities(String fromContext, String toContext, List ids)
 							    for (ToolConfiguration config: toolConfs) {
 								if (config.getToolId().equals(TOOL_ID)) {
 								    SitePage p = config.getContainingPage();
+									// Checking if the 'source' property of fromSite and toSite are same or not by replacing respective siteIds with a constant.
```
Indentation looks wrong.

ouit0408 committed:
WL-4200 Web Content links in template sites are not duplicated in new site.

Checking if the web content url has already been added in new site.if
clause was checking for webcontent urls which never were equal because of
the unique siteId for each site.introduced a constant to check for
equality .

ouit0408 commented:
Have pushed the changes.
